### PR TITLE
TV Detail screen implementation

### LINKED ISSUE #120: [FR]: TV Detail screen
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Related app

TV - Android

### Describe the problem

There aren't any details for movies in the TV application.

### Describe the solution

I am implementing the details screen, especially for the TV application.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Ramink78**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly.

Take a look at the [contributing guidelines](https://github.com/moallemi/Film-Time/blob/dev/CONTRIBUTING.md) for this project.

### Checklist
- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the linter (code style) passes
- [x] Appropriate docs were updated (if necessary)

Fixes #120

### CODE REVIEW

File: tv/src/main/java/io/filmtime/tv/FilmTimeTvApp.kt [hunk 1]
```diff
@@ -41,61 +46,49 @@ fun FilmTimeTvApp() {
...
   TvNavHost(
     navController = navController,
     modifier = Modifier.offset {
       IntOffset(0, navHostYOffset)
     },
-    onTopBarVisibleChange = { isVisible ->
-      isTopBarVisible = isVisible
-    },
+    onHideTabBar = { isTopBarVisible = false },
+    onShowTabBar = { isTopBarVisible = true },
```
Comment (moallemi) [inline tv/src/main/java/io/filmtime/tv/FilmTimeTvApp.kt]: Why do we need to have two lambdas for a boolean variable? Why the previous approach does not work?

File: tv/src/main/java/io/filmtime/tv/ui/navigation/TvNavHost.kt [hunk 2]
```diff
@@ -15,15 +16,26 @@ import io.filmtime.tv.ui.settings.settingsGraph
...
     modifier = modifier,
   ) {
     homeGraph(
-      onTopBarVisibleChange = onTopBarVisibleChange,
+      onFirstItemVisibleChange = { isVisible ->
+        if (isVisible) {
+          onShowTabBar()
+        } else {
+          onHideTabBar()
+        }
```
Comment (moallemi) [inline tv/src/main/java/io/filmtime/tv/ui/navigation/TvNavHost.kt]: Why do we need this logic here? why do not we only hoist the firstVisibilityChange to parent? (Related to my previous comment [here](https://github.com/moallemi/Film-Time/pull/121/files#diff-8f59a283ddb758a020022fe846b90f398fd99e60b61c5d9ea171c4f2b2e1f1dcR91))

File: tv/src/main/java/io/filmtime/tv/ui/home/HomeScreen.kt [hunk 3]
```diff
@@ -23,24 +23,25 @@ import io.filmtime.tv.ui.component.MoviesRow
 
 @Composable
 fun HomeScreen(
-  onTopBarVisibleChange: (Boolean) -> Unit,
+  onFirstItemVisibleChange: (isVisible: Boolean) -> Unit,
+  onNavigateToMovieDetail: (tmdbId: Int) -> Unit,
```
Comment (moallemi) [inline tv/src/main/java/io/filmtime/tv/ui/home/HomeScreen.kt]: 💅💅💅 Why not `onMovieDetailClick`? maybe parent Composable does not want to navigate and only wants to show a message 😄

File: tv/src/main/java/io/filmtime/tv/ui/home/HomeScreen.kt [hunk 4]
```diff
@@ -70,6 +72,7 @@ fun HomeScreenContent(
         MoviesRow(
           thumbnails = it.items,
           title = it.title,
+          onClick = onNavigateToMovieDetail,
         )
       }
```
Comment (moallemi) [inline tv/src/main/java/io/filmtime/tv/ui/home/HomeScreen.kt]: We are using the name `MoviesRow` for displaying a row of `VideoThumbnail`s which consist of both Movies and Show. 
a) We can rename it to `VidoeThumbnailsRow` or `VidoeThumbnailsSection` or whatever meaningful 

b) There is a bug here that we are using `onNavigateToMovieDetail` for both movie and shows. 
Maybe we can add the type check logic inside MoviesRow or here in home screen:

Approach 1:
```kotlin
MoviesRow(
   onShowClick: (tmdbId) -> Unit,
   onMovieClick: (tmdbId) -> Unit,
)
```

Approach 2:
```kotlin
MoviesRow(
   onItemClick: (type, tmdbId) -> Unit,
)
```
then in HomeScreen:
```kotlin
MoviesRow(
 onItemClick = { type, tmdbId ->
 if type == video => onNavigateToMovieDetail()
if type == show => { // TODO: #ISSUE_NUMBER we will implement show detail screen in another issue}
}
)
```

File: tv/src/main/java/io/filmtime/tv/ui/detail/movie/MovieDetailSceen.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          .fillMaxSize(),
+        contentPadding = PaddingValues(vertical = 100.dp),
+        verticalArrangement = Arrangement.spacedBy(25.dp),
+      ) {
+        item { headerContent(detail) }
+        detail.ids.tmdbId?.let {
+          item { castsContent(it) }
+          item { similarContent(it) }
+        }
+        item { informationContent(detail) }
```
Comment (moallemi) [inline tv/src/main/java/io/filmtime/tv/ui/detail/movie/MovieDetailSceen.kt]: Better to add key to items for improving performance.

```suggestion
        item(
        key = "header",
        ) { headerContent(detail) }
        detail.ids.tmdbId?.let {
          item(
          key = "casts",
          ) { castsContent(it) }
          ...
```

File: tv/src/main/java/io/filmtime/tv/ui/detail/movie/DetailNavigation.kt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.ui.Modifier
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.composable
+import kotlinx.serialization.Serializable
+
+@Serializable
+private data class MovieDetail(val video_id: Int)
```
Comment (moallemi) [inline tv/src/main/java/io/filmtime/tv/ui/detail/movie/DetailNavigation.kt]: Why not `videoId`?

**Review by moallemi — COMMENTED** [review]
Dear @Ramink78 Thanks for your contribution 🙏

File: tv/src/main/java/io/filmtime/tv/FilmTimeTvApp.kt (hunk 1 above)
Comment (Ramink78) [reply tv/src/main/java/io/filmtime/tv/FilmTimeTvApp.kt]: Yes, you are right. It can be better.

**Pull request merged**